lucille-core/src/main/java/com/kmwllc/lucille/indexer/SolrIndexer.java

```diff
@@ -152,6 +152,7 @@ protected void sendToIndex(List<Document> documents) throws Exception {
 
         if (solrDocRequests.containsIdForDeletion(solrId)) {
           sendDeletionBatch(collection, solrDocRequests.getDeleteIds());
+          solrDocRequests.resetDeletes();
```

Comment:
Good catch here. Were any tests failing when this was omitted? If not, I suggest flexing this in a test. Thinking out loud -- we'd need a delete, then an add (to cause the deletions to be sent), then another delete which should be for a _different_ doc. Then when deletes are sent the second time, need to make sure the delete for the first doc isn't issued again...

Reply:
Both of the two most recently added tests "testSolrIndexerUnrelatedAddAddDeleteAdd" and "testSolrIndexerDeleteAddUnrelatedDelete" will fail if the line to resetDeletes is removed.